delphi-checks/src/main/java/au/com/integradev/delphi/checks/PublicFieldCheck.java

```diff
@@ -43,6 +43,8 @@ public DelphiCheckContext visit(FieldDeclarationNode field, DelphiCheckContext c
   }
 
   private static boolean isRecordField(FieldDeclarationNode field) {
-    return field.getFirstParentOfType(TypeDeclarationNode.class).isRecord();
+    TypeDeclarationNode typeParent = field.getFirstParentOfType(TypeDeclarationNode.class);
```

Comment:
Do `TypeNode` instead and you won't need to do any null-checking - it should always be present.